templates/common/_base/files/configure-ovs-network.yaml

```diff
@@ -0,0 +1,313 @@
+    # are not supported.
+
+    parse_yaml() {
+      local yaml_file=$1
+      local prefix=$2
+      local s
+      local w
+      local fs
+
+      s='[[:space:]]*'
+      w='[a-zA-Z0-9_.-]*'
+      fs="$(echo @|tr @ '\034')"
+
+      (
+        sed -e '/- [^\“]'"[^\']"'.*: /s|\([ ]*\)- \([[:space:]]*\)|\1-\'$'\n''  \1\2|g' |
+
+        sed -ne '/^--/s|--||g; s|\"|\\\"|g; s/[[:space:]]*$//g;' \
+            -e "/#.*[\"\']/!s| #.*||g; /^#/s|#.*||g;" \
+            -e "s|^\($s\)\($w\)$s:$s\"\(.*\)\"$s\$|\1$fs\2$fs\3|p" \
+            -e "s|^\($s\)\($w\)${s}[:-]$s\(.*\)$s\$|\1$fs\2$fs\3|p" |
+
+        awk -F"$fs" '{
+            indent = length($1)/2;
+            if (length($2) == 0) { conj[indent]="+";} else {conj[indent]="";}
+            vname[indent] = $2;
+            for (i in vname) {if (i > indent) {delete vname[i]}}
+                if (length($3) > 0) {
+                    vn=""; for (i=0; i<indent; i++) {vn=(vn)(vname[i])("_")}
+                    printf("%s%s%s%s=(\"%s\")\n", "'"$prefix"'",vn, $2, conj[indent-1],$3);
+                }
+            }' |
+
+        sed -e 's/_=/+=/g' |
+
+        awk 'BEGIN {
+                FS="=";
+                OFS="="
+            }
+            /(-|\.).*=/ {
+                gsub("-|\\.", "_", $1)
+            }
+            { print }'
+      ) < "$yaml_file"
+    }
+
+    unset_variables() {
+      # Pulls out the variable names and unsets them.
+      local variable_string="$@"
+      unset variables
+      variables=()
+      for variable in ${variable_string[@]}; do
+        variables+=($(echo $variable | grep '=' | sed 's/=.*//' | sed 's/+.*//'))
+      done
+      for variable in ${variables[@]}; do
+        unset $variable
+      done
+    }
+
+    create_variables() {
+      local yaml_file="$1"
+      local prefix="$2"
+      local yaml_string="$(parse_yaml "$yaml_file" "$prefix")"
+      unset_variables ${yaml_string[@]}
+      eval "${yaml_string}"
+    }
+
+    CONFIG_FILE="/etc/sysconfig/network-scripts/ovs-config.yaml"
+
+    iface=""
+    counter=0
+    # find default interface
+    while [ $counter -lt 12 ]; do
+      iface=$(ip -j route show default | jq -r '.[0].dev')
+      if [ -n "$iface" ]; then
+        echo "Default gateway interface found: ${iface}"
+        break
+      fi
+      counter=$((counter+1))
+      echo "No default route found on attempt: ${counter}"
+      sleep 5
+    done
+
+    if [ "$iface" = "br-ex" ]; then
+      echo "Networking already configured and up for br-ex!"
+      exit 0
+    fi
+
+    phys_port_opts=""
+    CONFIG_PROVIDED=false
+    BONDING=false
+    # check if config file is provided at /etc/sysconfig/network-scripts/ovs-config.yaml
+    if [ -f "$CONFIG_FILE" ]; then
+      create_variables ${CONFIG_FILE} ovs_
+      echo "Variables parsed from OVS config file: $(compgen -A variable | grep ovs_)"
+
+      if [ ${#ovs_interfaces__name[@]} -gt 2 ]; then
```

Comment:
Should this be validating that it has one of each i.e. the physical NIC and the bridge name? It just checks for the length of the array being > 2. But doesn't validate the two entries per-se? Am I missing something?

Reply:
it's just a simple validation to ensure the number of entries are 2. I.E. someone doesn't write an nmstate config file expecting this to also handle other bonds or linux interfaces. You are right there could be more validation here, but given the time constraint and NM bugs this is what I got to finally pass. We can enhance it more in a follow up patch.